### PR TITLE
Finished Habit Tracker console app

### PR DESCRIPTION
**Pull request by ngalantino**

### CODE REVIEW

File: README.md
Comment (chrisjamiecarter) [inline README.md:1]: Could you please put this file inside the project folder.

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @ngalantino 👋 ,

Are you able to move the following files to inside your project directory?
It allows me to review your code at the same time as others.

image

Thanks,
Chris

File: README.md
Comment (ngalantino) [reply README.md:1]: Hi @chrisjamiecarter -

Thanks for reviewing my code!

Comment (chrisjamiecarter) [reply README.md:1]: Perfect, I will now continue my review! 👏

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @ngalantino 👋 ,

Fantastic work on this application!

Well done on the README and you code writing style.

I have added some minor comments to help support your growth as a developer. You do not need to action these now, just digest them.

Unfortunately, I did spot a bug within your application.
When I load the application and attempt to view all records I get the following error:
image

Can you revisit to understand and fix the issues?

I know what the issue is, so please feel free to reach out if you have any questions or if you'd like to collaborate on this.

Once you have resolved, commit and push your changes and I will continue my review.

Thanks,
@chrisjamiecarter

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @ngalantino 👋 ,

Great job on getting the comments I made addressed 🎉 !

Unfortunately the bug is still not resolved... 😦 

Note the below code:

image

Your code to create the table is in a constructor
`public Program()`

However, if you put a break point here, you will see it is never hit. This is because in .NET, the entry point for a console application is the Main method, which is a static method.

My recommendation would be to move the constructor code into an `InitialiseDatabase()` method and call that before `GetUserInput()` in `Main()`.

Thanks
@chrisjamiecarter

**Comment by chrisjamiecarter** [comment]
@ngalantino  If you delete your database file, and then re=run your app you should see the same results.

**Comment by ngalantino** [comment]
@chrisjamiecarter - I think I've got it now. I didn't know that console apps don't execute the constructor code!  Thanks for pointing that out!

File: HabitTracker.ngalantino/HabitTracker.ngalantino/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "profiles": {
+    "HabitTracker.ngalantino": {
+      "commandName": "Project",
+      "workingDirectory": "C:\\Users\\Shadow\\source\\CodeReviews.Console.HabitTracker\\HabitTracker.ngalantino\\HabitTracker.ngalantino\\"
```
Comment (chrisjamiecarter) [inline HabitTracker.ngalantino/HabitTracker.ngalantino/Properties/launchSettings.json:5]: Just an FYI that I had to change this value for my machine in order for it to run.
Make sure that if you include hardcoded paths like this that you document it in the README file under an Installation or Usage section.

**Review by chrisjamiecarter — APPROVED** [review]
Hey @ngalantino ,

Great effort and persistence in getting these comments resolved.

Congratulations on another application submitted 🎉 

A few tips in regards to the user experience:
- You could clear the console each time before you write the main menu. This would allow the user to segment an action, maintain readability of the console window and improve the experience for the user.
- If a user chooses the `View All Records` option, put a `Console.WriteLine("Press any key to continue...");` and a `Console.Read();` after you have displayed the data. This will pause the screen, allow the user to read/digest the information, and then when they are ready, go back to the main menu (which as the point above, will clear the screen and write a fresh main menu).

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points.

You're doing a fantastic job, and these are just a few suggestions to support your growth as a developer.

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter

**Pull request closed without merging**